lib/rules/utils/ast-utils.js

```diff
@@ -1501,6 +1501,37 @@ module.exports = {
         return directives;
     },
 
+    /**
+     * Checks whether a specified node is an ExpressionStatement that is either part of,
+     * or immediately follows a (possibly empty) directive prologue.
+     * @see {@link http://www.ecma-international.org/ecma-262/6.0/#sec-directive-prologues-and-the-use-strict-directive}
+     * @param {SourceCode} sourceCode The source code object to get tokens.
+     * @param {ASTNode} node An expression node.
+     * @returns {boolean} Whether a specified node is an ExpressionStatement that is either part of,
+     * or immediately follows a (possibly empty) directive prologue.
+     */
+    isExpressionInOrJustAfterDirectivePrologue(sourceCode, node) {
```

Comment:
This is a refactoring of the deleted function `isPartOfDirectivePrologue` from `quotes.js` with few changes:
* Renamed the function to `isExpressionInOrJustAfterDirectivePrologue` to better describe what it does
* Changed the argument to the be the parent node (the `ExpressionStatement`)
* Added a check to make sure that the specified node is really an `ExpressionStatement`
* Fixed to logic that checks if a sibling node is a directive to ensure that literals in parentheses test negative
* Added an additional `SourceCode` argument to the function (necessary to test for parentheses)